step-release-vis/src/app/services/environment.ts

```diff
@@ -15,33 +14,33 @@ export class EnvironmentService {
   }
 
   // xs: 0-100, ys: timestamps
-  // TODO(naoai): write test
   getPolygons(jsonFile: string): Observable<Polygon[]> {
     return this.readJson(jsonFile)
       .pipe(map(environments => this.calculatePolygons(environments)));
   }
 
+  // TODO(naoai): compute the coordinates for the polygons
   private calculatePolygons(environments: Environment[]): Polygon[] {
     return [];
   }
 
-  private extractPoints(candsInfo: CandidateInfo[]): CandidatesSnapshot {
-    return [];
-  }
+  private getPercentages(candsInfo: CandidateInfo[]): Map<string, number> {
+    const candInfo2percentage: Map<string, number> = new Map();
+    let totalJobSum = 0;
 
-  // TODO(ancar): calculate the percentages for cands at given timestamp
-  private getPercentages(candsInfo: CandidateInfo[]): CandidatesSnapshot {
-    return [];
+    for (const candInfo of candsInfo) {
+      totalJobSum += candInfo.job_count;
+    }
+
+    for (const candInfo of candsInfo) {
+      const percentage = candInfo.job_count / totalJobSum * 100;
```

Comment:
optional: I am also fine w having a [0,1] instead of [0,100] range, ppl can understand that too